app/models/katello/content_view_version.rb

```diff
@@ -240,12 +240,22 @@ def check_ready_to_promote!
     end
 
     def check_ready_to_destroy!
+      check_used_by_composite!
+
       if environments.any? && !organization.being_deleted?
         fail _("Cannot delete version while it is in environments: %s") % environments.map(&:name).join(",")
       end
       return true
     end
 
+    def check_used_by_composite!
+      if composite_content_views.any? && !organization.being_deleted?
+        fail _("Cannot delete version while it is in use by composite content views: %s") %
```

Comment:
I feel like these types of things would be better in two functions: one that checks the status, and one that then calls the former and throws a failure if it cannot. Something like `in_compose?` or `belongs_to_composite?` and then another function that I am having a hard time naming at present because I think what I find strange is that we do this check in the action instead of it being done when `destroy!` is called.